### PR TITLE
ci: remove redundant 'cd build'; use 'set -e' in scripts

### PR DESCRIPTION
**Pull request by umarcor**
This PR  brings a couple of minor enhancements to the CI scripts.

### CODE REVIEW

**Comment by umarcor** [comment]
I think that the removal of `cd build` in this PR does fix the "empty" commit SHA in the latest updates to https://github.com/SymbiFlow/symbiflow.github.io/commits/master.

File: .github/workflows/build-and-deploy.yml [hunk 1]
```diff
@@ -39,14 +39,14 @@ jobs:
...
       - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
       - name: Deploy website
         if: github.event_name == 'push'
         run: |
           git config --global user.name SymbiFlow
           git config --global user.mail robot@mith.ro
           git config --global push.default simple
-          git fetch origin --tags
```
Comment (mithro) [inline .github/workflows/build-and-deploy.yml:48]: We want `--tags` so that `git describe` gives the right output.

Comment (umarcor) [reply .github/workflows/build-and-deploy.yml:48]: Option `fetch-depth: 0` was implemented for this use case. See:

https://github.com/actions/checkout/#usage
```
    # Number of commits to fetch. 0 indicates all history for all branches and tags.
    # Default: 1
    fetch-depth: ''
```

and https://github.com/actions/checkout/issues/250#issuecomment-634799150.

**Comment by umarcor** [comment]
Closing because of https://github.com/f4pga/f4pga-website/tree/main/.github/workflows.

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/f4pga/f4pga-website">f4pga-website/.github/workflows at main · f4pga/f4pga-website</a></strong></div><div>Contribute to f4pga/f4pga-website development by creating an account on GitHub.</div></blockquote>

**Pull request closed without merging**